oscarapi/views/utils.py

```diff
@@ -54,3 +55,26 @@ def check_basket_permission(self, request, basket_pk=None, basket=None):
             basket = generics.get_object_or_404(Basket.objects, pk=basket_pk)
         self.check_object_permissions(request, basket)
         return basket
+
+
+class CustomPageNumberPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = "page_size"
+    max_page_size = 10000
+    page_query_param = "page"
+
+    def get_next_link(self):
+
+        if not self.page.has_next():
+            return None
+        url = self.request.build_absolute_uri()
+        page_number = self.page.next_page_number()
+
+        return replace_query_param(url, self.page_query_param, page_number)
+
+    def get_previous_link(self):
+        if not self.page.has_previous():
+            return None
+        url = self.request.build_absolute_uri()
+        page_number = self.page.previous_page_number()
+        return replace_query_param(url, self.page_query_param, page_number)
```

Comment:
Why is this needed exactly?

I can see that you solved the problem with assining the strategy to the entire queryset. But I dn't understand how this has anything to do with that.

Reply:
By creating pagination for this endpoint (and limiting the amount of result) we also need to build the next/prev pages of the response